itest/chantools_harness.go

```diff
@@ -0,0 +1,160 @@
+package itest
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil/psbt"
+	"github.com/stretchr/testify/require"
+)
+
+// chantoolsCmdBuilder is a helper function that creates a new exec.Cmd instance
+// for the chantools binary.
+func chantoolsCmdBuilder(binPath string, workingDir string,
+	args ...string) exec.Cmd {
+
+	cmd := exec.Command(binPath, args...)
+
+	// Set the working directory so that any log files are handled
+	// gracefully.
+	cmd.Dir = workingDir
+
+	// Set the wallet password.
+	cmd.Env = append(os.Environ(), "WALLET_PASSWORD=-")
+
+	// Set the seed generation passphrase.
+	cmd.Env = append(cmd.Env, "AEZEED_PASSPHRASE=-")
+
+	return *cmd
+}
+
+// ChantoolsHarness is a helper struct that provides a way to interact with
+// the chantools binary.
+type ChantoolsHarness struct {
+	// path is the path to the chantools binary.
+	path string
+
+	// workingDir is the chantools harness working directory.
+	workingDir string
+
+	// walletDbPath is the path to the wallet.db file created by chantools.
+	walletDbPath string
+}
+
+// NewChantoolsHarness creates a new instance of the ChantoolsHarness struct.
+func NewChantoolsHarness(t *testing.T) ChantoolsHarness {
```

Comment:
Very cool! Awesome to have this tested automatically.
Makes me think we should probably expose these functions in `chantools` so we can use them directly from Go instead of needing to invoke the binary.

Reply:
While you were away, I attempted to expose those functions in `chantools`, but it turned out to be a significant change that interfered with our short-term priorities.

I agree that exporting that functionality from `chantools` is a good idea. However, one advantage of the command-line harness is that it more closely mimics typical user behavior.

Reply:
If we choose to import chantools as a library here we can don't need the chantools check in the  Makefile target anymore.

Reply:
Yes would be nice to not need another external binary. As long as the exported functions correspond to the CLI commands correctly.